modules/nix/nix-optimise.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, ...}:
+with lib;
+
+let
+  cfg = config.nix.optimise;
+in
+
+{
+  options = {
+    nix.optimise = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = "Automatically run the nix store optimiser at a specific time.";
+      };
+      periodSeconds = mkOption {
+        default = 60 * 60 * 8;
+        type = types.int;
+        description = "Interval between runs of the optimizer in seconds.";
+      };
```

Comment:
The upstream NixOS module uses absolute time rather than a recurring period to schedule this. Unless there's some reason to diverge here, we should match that behaviour and its 03:45 time; perhaps rename this to `intervals` (to match `nix.gc.interval`), use it as `StartCalendarInterval` in the launchd daemon, and set the default to `[{ Hour = 3; Minute = 45; }]`. (You can set the type to `types.listOf types.attrs`; it's too much of a pain to try fishing out the exact correct type from the `launchd` options as things stand, unfortunately.)

A potential issue here is that, although launchd will run timers missed due to system sleep on wake, it will not do so if the computer was turned entirely off during the scheduled time. The same issue applies already to `nix.gc.interval` and `StartInterval` in general, though, and using a list allows people to set multiple times per day if they are worried about that. You could add a note in the description about this if you want, but you don't have to.

(Additional, very minor nit: we should say "optimiser" here to match the setting name, as Nix uses standard British English spelling.)

Reply:
I just saw https://github.com/LnL7/nix-darwin/pull/716 - I'd like to amend my suggestion to naming it `interval` for consistency, and if that PR is merged before this one, it should be easier to specify the correct type.